enhancements/insights/pulling-data-from-ocm.md

```diff
@@ -0,0 +1,139 @@
+
+### Risks and Mitigations
+
+#### OCM API is down
+
+Risk: OCM API is down or doesn't provide up to date data.
+
+Risk: Insights Operator is unable to expose/update the data in the OpenShift API
+
+Mitigation: Introduce a new state in the Insights Operator (e.g "SCADataDegraded") and
+create a new alert based on this new state.
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the data from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config-managed` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### Data in API
+
+The SCA certificate is available via the `etc-pki-entitlement` secret in the `openshift-config-managed` namespace.
+
+### Update period
+- Insights Operator query the OCM API every 8 hours and downloads the full data provided
+
+### Test Plan
+
+- `insights-operator-e2e-tests` suite can verify the SCA cert data
+  is available
+- Basic test of the validity of the SCA certs. Mount the `etc-pki-entitlement` secret and run e.g `yum install` in the container
+
+### Graduation Criteria
+
+This feature is planned as a technical preview in OCP 4.9 and is planned to go GA in 4.10.
+
+#### Dev Preview -> Tech Preview
+- opt-in feature (called `InsightsOperatorPullingSCA`) enabled with `TechPreviewNoUpgrade` feature set
+- Insights Operator is able to download the data from OCM API and expose it in a cluster API
+- basic functionality is tested
+- this new functionality is documented
+
+#### Tech Preview -> GA
+- ability to distinguish various error states - e.g organization doesn't have SCA allowed versus API returns an error
+- inform a cluster user about the error state (problem with pulling the certificates)
+- the feature might be moved to a different OCP component
+
+#### Removing a deprecated feature
+
+The periodical data pulling can be easily disabled in the cluster configuration. Removing this feature will require updating the Insights operator code base and will remove the `etc-pki-entitlement` secret from the `openshift-config-managed` namespace.
+
+### Upgrade / Downgrade Strategy
+
+There is no upgrade/downgrade strategy needed.
+
+### Version Skew Strategy
+
+There is no Skew strategy needed. This work should have no impact on the upgrade. It doesn't require any coordinated behavior in the control plane. No other components will change.
+
+## Implementation History
+
+There are no other major milestones in the implementation history than the graduation criteria mentioned above.
+
+## Drawbacks
+
+There is no significant drawback.
+
+## Alternatives
+
+- Alternative is to implement this functionality in another control plane component/operator (e.g openshift-controller-manager).
+- Current state, which is the manual addition of the SCA certs to cluster worker nodes. This is not very convenient because the SCA certs change regularly and the change requires node reboot.
```

Comment:
Will the automated update somehow avoid those reboots? How often do the certs change and what's the impact of rebooting every OpenShift cluster that often? Remember, large metal clusters can take days to fully reboot.

Reply:
the implementation is going to make the token available as a secret in a namespace in the cluster, so it won't go onto the nodes themselves.

Not exactly clear how consumers will access the secret, however, without using something like the (still being developed) csi shared resource driver.

Reply:
OK, I don't understand the comment on line 138 about manually adding the certs to worker nodes, then.

Reply:
yeah i think that's an incomplete statement.  Today what people do is put the token on the worker node and then configure crio to automount it, i believe.  

As far as i can tell, this EP isn't really a full replacement for that approach because it appears to only address "how do i get the token to my cluster" but not "how does a workload consume the token from the cluster"  (which is fine, i'm not saying this EP has to be a full replacement for that process, but it probably needs to be clearer what this EP does and does not provide, and what is still left on the admin or user to do, once this EP is delivered)